jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MimeTypes.java

```diff
@@ -222,145 +206,33 @@ public Type getBaseType()
-                    props.stringPropertyNames().stream()
-                        .filter(t -> t != null)
-                        .forEach(t ->
-                        {
-                            String charset = props.getProperty(t);
-                            if (charset.startsWith("-"))
-                                __assumedEncodings.put(t, charset.substring(1));
-                            else
-                                __inferredEncodings.put(t, props.getProperty(t));
-                        });
-
-                    if (__inferredEncodings.isEmpty())
-                    {
-                        LOG.warn("Empty encodings at {}", resourceName);
-                    }
-                    else if ((__inferredEncodings.size() + __assumedEncodings.size()) < props.keySet().size())
-                    {
-                        LOG.warn("Null or duplicate encodings in resource: {}", resourceName);
-                    }
-                }
-                catch (IOException e)
-                {
-                    if (LOG.isDebugEnabled())
-                        LOG.warn("Unable to read encoding resource: {}", resourceName, e);
-                    else
-                        LOG.warn("Unable to read encoding resource: {} - {}", resourceName, e.toString());
-                }
-            }
-        }
-        catch (IOException e)
-        {
-            if (LOG.isDebugEnabled())
-                LOG.warn("Unable to load encoding resource: {}", resourceName, e);
-            else
-                LOG.warn("Unable to load encoding resource: {} - {}", resourceName, e.toString());
-        }
-    }
-
-    private final Map<String, String> _mimeMap = new HashMap<String, String>();
-
-    /**
-     * Constructor.
-     */
     public MimeTypes()
     {
+        this(DEFAULTS);
     }
 
-    public Map<String, String> getMimeMap()
+    public MimeTypes(MimeTypes defaults)
     {
-        return _mimeMap;
-    }
-
-    /**
-     * @param mimeMap A Map of file extension to mime-type.
-     */
-    public void setMimeMap(Map<String, String> mimeMap)
-    {
-        _mimeMap.clear();
-        if (mimeMap != null)
+        if (defaults != null)
         {
-            for (Entry<String, String> ext : mimeMap.entrySet())
-            {
-                _mimeMap.put(StringUtil.asciiToLowerCase(ext.getKey()), normalizeMimeType(ext.getValue()));
-            }
+            _mimeMap.putAll(defaults.getMimeMap());
+            _assumedEncodings.putAll(defaults.getAssumedMap());
+            _inferredEncodings.putAll(defaults.getInferredMap());
         }
     }
 
     /**
      * Get the MIME type by filename extension.
-     * Lookup only the static default mime map.
      *
      * @param filename A file name
      * @return MIME type matching the longest dot extension of the
      * file name.
      */
-    public static String getDefaultMimeByExtension(String filename)
+    public String getMimeByExtension(String filename)
```

Comment:
We should have a `getMimeByPath(Path path)` and which can use the fallback JVM mechanism.

https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/nio/file/Files.html#probeContentType(java.nio.file.Path)

Reply:
I think it is probably a good idea to use the prob somewhere in our resource handling, but I'm not sure that doing file ops in `MimeTypes` is the right way to go.   Maybe, but let's do that when we actually start using it. 

@lachlan-roberts can you think about this in your ResourceService work.